### PR TITLE
fix(API): enhance hostname and name check

### PR DESCRIPTION
**Pull request by sc979**
## Description

Allow spaces on host
More accurate check for RFC compliant check

**Fixes** # (MON-6093)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 19.04.x
- [ ] 19.10.x
- [ ] 20.04.x
- [x] 20.10.x (master)

### CODE REVIEW

File: www/include/Administration/parameters/DB-Func.php [hunk 1]
```diff
@@ -1051,7 +1051,7 @@ function updateRemoteAccessCredentials($db, $form, $centreonEncryption): void
     unset($ret['apiCredentials']);
 
     //convert values
-    $ret['apiSelfSignedCertificate'] = $ret['apiSelfSignedCertificate'] == 1 ? 'yes' : 'no';
+    $ret['apiPeerValidation'] = (int) $ret['apiPeerValidation'] === 1 ? 'no' : 'yes';
```
Comment (kduret) [inline www/include/Administration/parameters/DB-Func.php:1054]: `1` means `no`?

Comment (sc979) [reply www/include/Administration/parameters/DB-Func.php:1054]: it's odd indeed. When checked the value is '1' for -> 
image

But when checked, "allow self signed certificate" means do not check "peer validation"
so yes, 1 means no in DB :)

Comment (kduret) [reply www/include/Administration/parameters/DB-Func.php:1054]: ok :)
so the name of the form input is not good

Comment (kduret) [reply www/include/Administration/parameters/DB-Func.php:1054]: let's update this later

**Review by loiclau — APPROVED** [review]

**Pull request merged**